### PR TITLE
 add non-default routes to handle security gateways

### PR DESCRIPTION
**Pull request by eriknordmark**
When application instances want to connect to an IP address on the local subnet the PBR code used to always send to the default router. Some security gateway configurations do not allow such routing back to the same interface. Hence here we add all of the routes into the PBR table for the bridge.

### CODE REVIEW

File: pkg/pillar/cmd/zedrouter/pbr.go [hunk 1]
```diff
@@ -72,27 +73,30 @@ func PbrRouteAddDefault(bridgeName string, port string) error {
...
-	}
-	log.Infof("PbrRouteAddDefault(%s, %s) adding %v\n",
-		bridgeName, port, myrt)
-	if err := netlink.RouteAdd(&myrt); err != nil {
-		errStr := fmt.Sprintf("Failed to add %v to %d: %s",
-			myrt, myrt.Table, err)
-		log.Errorln(errStr)
-		return errors.New(errStr)
+	for _, rt := range routes {
+		MyTable := FreeTable + ifindex
```
Comment (gkodali-zededa) [inline pkg/pillar/cmd/zedrouter/pbr.go]: A nit. I guess this line can be outside loop.

File: pkg/pillar/cmd/zedrouter/pbr_linux.go [hunk 2]
```diff
@@ -18,31 +18,21 @@ import (
...
 	routes, err := netlink.RouteListFiltered(syscall.AF_INET,
 		&filter, fflags)
 	if err != nil {
 		log.Fatalf("RouteList failed: %v\n", err)
 	}
-	log.Debugf("getDefaultIPv4Route(%d) - got %d matches\n",
+	log.Debugf("getAllIPv4Routes(%d) - got %d matches\n",
 		ifindex, len(routes))
-	for _, rt := range routes {
-		if rt.LinkIndex != ifindex {
```
Comment (gkodali-zededa) [inline pkg/pillar/cmd/zedrouter/pbr_linux.go]: What did this test do? We already have the ifindex passed as part of filter to RouteListFiltered. Why did we need this additional check?

Comment (eriknordmark) [reply pkg/pillar/cmd/zedrouter/pbr_linux.go]: I think it was part of a belt and suspenders check; could have been that the original code didn't pass the OIF match.

File: pkg/pillar/cmd/zedrouter/pbr.go (hunk 1 above)
Comment (eriknordmark) [reply pkg/pillar/cmd/zedrouter/pbr.go]: Fixed

**Pull request merged**